functions/helloworld/index.js

```diff
@@ -36,54 +38,105 @@ exports.helloWorld = function helloWorld (context, data) {
 exports.helloGET = function helloGET (req, res) {
   res.send('Hello World!');
 };
-// [END helloGET]
+// [END functions_helloworld_get]
 
-// [START helloHttp]
+// [START functions_helloworld_http]
 /**
  * HTTP Cloud Function.
  *
  * @param {Object} req Cloud Function request context.
  * @param {Object} res Cloud Function response context.
  */
 exports.helloHttp = function helloHttp (req, res) {
-  res.send('Hello ' + (req.body.name || 'World') + '!');
+  res.send(`Hello ${req.body.name || 'World'}!`);
 };
-// [END helloHttp]
+// [END functions_helloworld_http]
 
-// [START helloBackground]
+// [START functions_helloworld_background]
 /**
  * Background Cloud Function.
  *
- * @param {Object} context Cloud Function context.
- * @param {Object} data Request data, provided by a trigger.
+ * @param {object} event The Cloud Functions event.
+ * @param {function} The callback function.
  */
-exports.helloBackground = function helloBackground (context, data) {
-  context.success('Hello ' + (data.name || 'World') + '!');
+exports.helloBackground = function helloBackground (event, callback) {
+  callback(null, `Hello ${event.data.name || 'World'}!`);
 };
-// [END helloBackground]
+// [END functions_helloworld_background]
 
-// [START helloPubSub]
+// [START functions_helloworld_pubsub]
 /**
  * Background Cloud Function to be triggered by Pub/Sub.
  *
- * @param {Object} context Cloud Function context.
- * @param {Object} data Request data, provided by a Pub/Sub trigger.
+ * @param {object} event The Cloud Functions event.
+ * @param {function} The callback function.
  */
-exports.helloPubSub = function helloPubSub (context, data) {
-  console.log('Hello ' + (data.name || 'World') + '!');
-  context.success();
+exports.helloPubSub = function helloPubSub (event, callback) {
+  const pubsubMessage = event.data;
+  const name = pubsubMessage.data ? Buffer.from(pubsubMessage.data, 'base64').toString() : 'World';
+  console.log(`Hello ${name}!`);
+  callback();
 };
-// [END helloPubSub]
+// [END functions_helloworld_pubsub]
 
-// [START helloGCS]
+// [START functions_helloworld_storage]
 /**
  * Background Cloud Function to be triggered by Cloud Storage.
  *
- * @param {Object} context Cloud Function context.
- * @param {Object} data Request data, provided by a Cloud Storage trigger.
+ * @param {object} event The Cloud Functions event.
+ * @param {function} The callback function.
+ */
+exports.helloGCS = function helloGCS (event, callback) {
+  const file = event.data;
+  const isDelete = file.resourceState === 'not_exists';
+
+  if (isDelete) {
+    console.log(`File ${file.name} deleted.`);
```

Comment:
No, as in this sample the idea is that the user themselves would perform different actions based on the value of `isDelete`. Smashing the conditional down into a string eliminates that.